test/libsolidity/SemanticTest.cpp

```diff
@@ -212,20 +271,60 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 				);
 			}
 
-			bool outputMismatch = (output != test.call().expectations.rawBytes());
-			// Pre byzantium, it was not possible to return failure data, so we disregard
-			// output mismatch for those EVM versions.
-			if (test.call().expectations.failure && !m_transactionSuccessful && !m_evmVersion.supportsReturndata())
-				outputMismatch = false;
-			if (m_transactionSuccessful != !test.call().expectations.failure || outputMismatch)
-				success = false;
+			bytes expectationOutput;
+			if (test.call().expectations.builtin)
+			{
+				auto builtin = m_builtins[test.call().expectations.builtin->signature];
+				if (std::optional<bytes> builtinResult = builtin(*test.call().expectations.builtin))
+					expectationOutput = builtinResult.value();
+				else
+					test.setFailure(true);
+			}
+			else
+				expectationOutput = test.call().expectations.rawBytes();
 
-			test.setFailure(!m_transactionSuccessful);
+			bool outputMismatch = (output != expectationOutput);
+			if (test.call().kind == FunctionCall::Kind::Builtin)
+			{
+				if (outputMismatch)
+					success = false;
+			}
+			else
+			{
+				// Pre byzantium, it was not possible to return failure data, so we disregard
+				// output mismatch for those EVM versions.
+				if (test.call().expectations.failure && !m_transactionSuccessful && !m_evmVersion.supportsReturndata())
+					outputMismatch = false;
+				if (m_transactionSuccessful != !test.call().expectations.failure || outputMismatch)
+					success = false;
+				test.setFailure(!m_transactionSuccessful);
+			}
 			test.setRawBytes(std::move(output));
 			test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName()));
 		}
+
+		for (std::shared_ptr<TestHook>& hook: m_testHooks)
+			hook->afterFunctionCall(test);
+	}
+
+	// The artificialFunctionCall is an artificially created function call,
```

Comment:
Why do we need it?

Reply:
I was wondering about this as well

Reply:
I need this for the isoltest events to find out whether the last call where creating events. I can only check this if there is an artificial call added after the last call. I also think that this could be useful for future `TestHooks` as well.

Reply:
Maybe it can be simplified now, need to check.

Reply:
I think we really don't need this anymore. I will remove it. 

The initial version basically didn't had this verification step with `verifyFunctionCall(..)` and I did the checks within the `afterFunctionCall(..)` initially. Thats why I needed that artificial function call after the last call.  @chriseth great that you ask about this!